File: test/corpus/basic/02-text-styling/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	// Force TrueColor profile for consistent testing
	out := termenv.NewOutput(nil, termenv.WithProfile(termenv.TrueColor))
	
	// Test basic text styling
	bold := out.String("Bold Text").Bold()
	italic := out.String("Italic Text").Italic()
	underline := out.String("Underlined Text").Underline()
	combined := out.String("Bold Italic Red").Bold().Italic().Foreground(out.Color("#FF0000"))
	
	fmt.Println(bold)
	fmt.Println(italic)  
	fmt.Println(underline)
	fmt.Println(combined)
}

File: test/corpus/basic/02-text-styling/case.ts
import { newOutput, Profile, rgbColor, withProfile } from '../../../../src/index.js';

// Force TrueColor profile for consistent testing
const out = newOutput(process.stdout, withProfile(Profile.TrueColor));

// Test basic text styling
const bold = out.string('Bold Text').bold();
const italic = out.string('Italic Text').italic();
const underline = out.string('Underlined Text').underline();
const combined = out.string('Bold Italic Red').bold().italic().foreground(rgbColor('#FF0000'));

console.log(bold.toString());
console.log(italic.toString());
console.log(underline.toString());
console.log(combined.toString());
